python/lsst/ci/tsort.py

```diff
@@ -1,59 +1,103 @@
-#!/usr/bin/env python
-# -- mjuric: Taken from https://pypi.python.org/pypi/tsort and slightly modified
-# -- mjuric: LGPL licenced
-# -*- coding: utf-8 -*-
-
+from typing import Iterator, Iterable, List, Mapping, Set, Tuple
 
 class GraphError(Exception):
     pass
 
 
-def tsort(edges):
-    """topologically sort vertices in edges.
-    edges: list of pairs of vertices. Edges must form a DAG.
-           If the graph has a cycle, then GraphError is raised.
-    returns: topologically sorted list of vertices.
-    see http://en.wikipedia.org/wiki/Topological_sorting
+def to_dep_graph(edges: Iterable[Tuple[str, str]]) -> Mapping[str, Set[str]]:
+    """Takes an iterable collection of (object, dependency) pairs and returns
```

Comment:
Please use active voice for doc strings, e.g. "Convert an iterable collection...to a graph set". Similarly for all the other doc strings you added or changed (there's a batch below that start with "Takes").

Also please restore the rest of the original doc string. Types aren't sufficient. You lost valuable information when you deleted the description of `edges` and the wikipedia link. Again...please look for this in the other doc strings that you changed.

Reply:
This wording has been improved and the link has been added to the `toposort` function